site/hooks/useElementWidth/index.ts

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useCallback, useState, RefObject } from "react";
+
+export const useElementWidth = (elementRef: RefObject<HTMLElement>) => {
+  const [width, setWidth] = useState<number>(0);
+
+  const updateWidth = useCallback(() => {
+    if (elementRef && elementRef.current) {
+      const { width } = elementRef.current.getBoundingClientRect();
+      setWidth(width);
+    }
+  }, [elementRef]);
+
+  const firstUpdateWidth = useCallback(() => {
+    if (elementRef && elementRef.current) {
+      let { width } = elementRef.current.getBoundingClientRect();
+      width =
+        width -
+        parseInt(
```

Comment:
just a personal preference:

```suggestion
      const rect = elementRef.current.getBoundingClientRect();
      const width =
        rect.width -
        parseInt(
```